File: app/storage/file/mcp_config_store.py
"""MCP Server 配置文件存储。

布局：data/mcp_configs/{name}.json
"""

from __future__ import annotations

from pathlib import Path
from typing import Any

from app.config.settings import get_settings
from app.storage.file.base import list_json_ids, read_json, write_json_atomic


class MCPConfigStore:
    """将 MCP Server 配置序列化为 data/mcp_configs/{name}.json。"""

    def _path(self, name: str) -> Path:
        safe_name = name.replace("/", "_").replace("\\", "_")
        return get_settings().data_dir / "mcp_configs" / f"{safe_name}.json"

    def save(self, data: dict[str, Any]) -> None:
        write_json_atomic(self._path(data["name"]), data)

    def get(self, name: str) -> dict[str, Any] | None:
        return read_json(self._path(name))

    def delete(self, name: str) -> bool:
        path = self._path(name)
        if path.exists():
            path.unlink()
            return True
        return False

    def list_all(self) -> list[dict[str, Any]]:
        ids = list_json_ids(get_settings().data_dir / "mcp_configs")
        result = []
        for name in ids:
            data = self.get(name)
            if data:
                result.append(data)
        return result

File: miniagents-assembled/src/main/java/com/codex/miniagents/infrastructure/storage/file/McpConfigStore.java
package com.codex.miniagents.infrastructure.storage.file;

import com.codex.miniagents.config.MiniAgentsProperties;
import com.codex.miniagents.tools.model.McpConfigData;

import org.springframework.stereotype.Repository;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.ArrayList;
import java.util.List;
import java.util.Optional;

@Repository
public class McpConfigStore {
    private final Path baseDir;

    private final FileJsonSupport jsonSupport;

    public McpConfigStore(MiniAgentsProperties properties, FileJsonSupport jsonSupport) {
        this.baseDir = properties.getDataDir().resolve("mcp_configs");
        this.jsonSupport = jsonSupport;
    }

    private Path path(String name) {
        String safe = name.replace("/", "_").replace("\\", "_");
        return baseDir.resolve(safe + ".json");
    }

    public void save(McpConfigData mcpConfigData) {
        jsonSupport.writeJsonAtomic(path(mcpConfigData.getName()), mcpConfigData);
    }

    public Optional<McpConfigData> get(String name) {
        return jsonSupport.readJson(path(name), McpConfigData.class);
    }

    public boolean delete(String name) {
        Path p = path(name);
        try {
            return Files.deleteIfExists(p);
        } catch (IOException e) {
            throw new RuntimeException("Failed to delete MCP config: " + name, e);
        }
    }

    public List<McpConfigData> listAll() {
        List<McpConfigData> results = new ArrayList<>();
        for (String id : jsonSupport.listJsonIds(baseDir)) {
            get(id).ifPresent(results::add);
        }
        return results;
    }
}
